Round Robin Edits -JB (#206)
jaybee84/ml-in-rd: content/03.combining-datasets.md

## Constructing machine learning-ready rare disease datasets

High-throughput 'omic' data methods generate high-dimensional data (or data containing many features), regardless of the underlying disease or condition being assayed.
A typical rare disease dataset is comprised of a small number of samples [@doi:10.1186/s13023-020-01424-6].

> A lack of samples gives rise to the “curse of dimensionality” (i.e., few samples but many features), which can contribute to the poor performance of models [@doi:10.1038/nrc2294] (Figure {@fig:1}a).

Reviewer: Citing Fig 1A (assuming I'm in the right place with https://github.com/jaybee84/ml-in-rd/blob/a7c10073261bed4c95db1bebfe407b7f9ac96784/content/images/figures/pdfs/figure-1-combining-datasets.png) seems a bit odd here.

Author: I was still envisioning our heatmap figure here for some reason :(

> A lack of samples gives rise to the “curse of dimensionality” (i.e., few samples but many features), which can contribute to the poor performance of models [@doi:10.1038/nrc2294] (Figure {@fig:1}a).

Author: ```suggestion
A lack of samples gives rise to the “curse of dimensionality” (i.e., few samples but many features), which can contribute to the poor performance of models [@doi:10.1038/nrc2294].
```

More features often means increased missing observations (_sparsity_), more dissimilarity between samples (_variance_), and increased redundancy between individual features or combinations of features (_collinearity_) [@doi:10.1038/s41592-018-0019-x], all of which contribute to a challenges in ML implementation.

When small sample sizes compromise an ML model's performance, then two approaches can be taken to improve the outcome: 1) increase the number of samples to reduce sparsity, variance, and collinearity, 2) improve the quality of samples to account for sparsity, variance, and collinearity.

> In the first approach, appropriate training, evaluation, and held-out validation sets could be constructed by combining multiple small individual rare disease cohorts (Fig.1b, Box 1).

Reviewer: ```suggestion
In the first approach, appropriate training, evaluation, and held-out validation sets could be constructed by combining multiple small individual rare disease cohorts (Figure {@fig:1}a, Box 1).
```

> In the authors' experience, this is often a neccessity in case of rare diseases.

Reviewer: ```suggestion
In the authors' experience, this is often a necessity in case of rare diseases.
```

In combining datasets, special attention should be directed towards _data harmonization_ since data collection methods can differ from cohort to cohort.

> Without careful selection of aggregation methods, one may introduce technical variability into the combined dataset which can negatively impact the ML model's ability to learn or detect meaningful signal.

Reviewer: ```suggestion
Without careful selection of aggregation methods, one may introduce technical variability into the combined dataset that can negatively impact the ML model's ability to learn or detect meaningful signal.
```

Steps such as reprocessing the data using a single pipeline, using batch correction methods [@doi:10.1093/biostatistics/kxj037; @doi:10.1093/nar/gku864], and normalizing raw values [@doi:10.1186/gb-2010-11-3-r25] may be necessary to mitigate unwanted technical variability. (Figure {@fig:1}a)

> In the second approach, small but meaningfully generated datasets can greatly enhance the performance of ML models too.

Reviewer: ```suggestion
In the second approach, small but meaningfully generated datasets can greatly enhance the performance of ML models.
```

> In the second approach, small but meaningfully generated datasets can greatly enhance the performance of ML models too.
> Specifically, improving labeling of data is critical in accounting for sparsity and variance in the data. 

Reviewer: ```suggestion
In the second approach, small but well-annotated datasets can greatly enhance the performance of ML models too.
Improving labeling of data is critical in accounting for sparsity and variance in the data. 
```

> In our experience, collaboration with domain experts has proved to be critical in gaining insight into potential sources of variation in the datasets.

Reviewer: Otherwise we use "critical" two sentences in a row

```suggestion
In our experience, collaboration with domain experts has proved to be essential in gaining insight into potential sources of variation in the datasets.
```

An anecdotal example from the authors' personal experience: conversations with a rare disease clinician revealed that samples in a particular tumor dataset were collected using vastly different surgical techniques (laser ablation and excision vs. standard excision). 
This information was not readily available to non-experts, but was obvious to the clinician. 
Addition of this kind of important metadata or labels to the samples can greatly help ML models become more effective in extracting biologically relevant patterns.
Such instances underline the fact that continuous collaboration with domain experts and the sharing of well-annotated data is needed to generate robust datasets in the future.

How would one understand if a composite dataset has undergone proper harmonization and annotation?
Ideally, the structure in the composite dataset will be aligned with variables of interest, such as phenotype labels if available.
Instead, if the samples from the same cohort tend to group together regardless of phenotype, revisiting the construction of the dataset is warranted.
In the next section, we will discuss approaches that can aid in identifying and visualizing structure in datasets.

> ![Combining datasets to increase training data](content/images/figures/pdfs/figure-1-combining-datasets.pdf){#fig:1}

Reviewer: ```suggestion
![Combining datasets to increase training data](images/figures/pdfs/figure-1-combining-datasets.pdf){#fig:1}
```

Reviewer: But also, should this be 
```suggestion
![Combining datasets to increase training data](content/images/figures/pdfs/figure-1-combining-datasets.png){#fig:1}
```

For this (the Manubot) version
